### PR TITLE
Adds connect screen feature

### PR DESCRIPTION
**Pull request by lbkchen**
Adds main connect screen with intro and pins and veteran/org pop up boxes.

I think I'll add the two other screens (veteran profile & parter org profile) in another PR later.
img_2352

For GM:
connect-fast

### CODE REVIEW

**Comment by lbkchen** [comment]
added the fonts so it looks less arial-ugly
img_f5f8f62f1365-1

**Comment by lbkchen** [comment]
Last image lol this is just much nicer looking

img_2358

File: src/components/ConnectBox.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+import Button from '../components/Button';
+
+export default class ConnectBox extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      animationValue: new Animated.Value(0),
+
```
Comment (nikitr) [inline src/components/ConnectBox.js]: nit: remove extra line

File: src/components/ConnectPin.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  render() {
+    return (
+      <View
+        style={[styles.baseContainer]}
+      >
+        {this.renderPin()}
+        <Image
+          source={require('../../assets/images/photogenic.jpg')}
```
Comment (nikitr) [inline src/components/ConnectPin.js:41]: after we add a profile_img field to the veteran and PO, this will pull from that?

File: src/screens/SignupScreen.js [hunk 3]
```diff
@@ -91,6 +91,9 @@ export default class SignupScreen extends React.Component {
         },
       },
     };
+
+    this.signUp = this.signUp.bind(this);
```
Comment (nikitr) [inline src/screens/SignupScreen.js:95]: why was this added here if it's also defined in the constructor?

File: src/components/ConnectBox.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  }
+
+  onBoxClose() {
+    Animations.fade(this.state.animationValue).start(() => {
+      this.props.onClose();
+    });
+  }
+
+  getConnectionTitle() {
+    return this.props.connection.roles ?
```
Comment (nikitr) [inline src/components/ConnectBox.js:48]: the partnering org model doesn't have a function to serialize the roles enum so we may encounter a problem with this when we try to access roles for PO

File: src/components/ConnectPin.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    return this.props.pinType == 'parterOrg' ? (
+      <Image
+        source={require('../../assets/images/pin-green.png')}
+        style={styles.pin}
+      />
+    ) : (
+      <Image
+        source={require('../../assets/images/pin.png')}
+        style={styles.pin}
+      />
```
Comment (nikitr) [inline src/components/ConnectPin.js:30]: are we differentiating the PO vs veteran pins here, by color (green vs white)?

File: src/screens/ConnectScreen.js [hunk 6]
```diff
@@ -1,17 +1,249 @@
...
+          <ConnectPin pinType="parterOrg" />
+        </MapView.Marker>
+      );
+    })
+  }
+
+  renderConnectBox() {
+    return this.state.activeConnection ? (
+      <ConnectBox
+        connection={this.state.activeConnection}
```
Comment (nikitr) [inline src/screens/ConnectScreen.js:208]: is activeConnection also an object (PO or veteran) which is the veteran/org that has currently been focused on?

File: src/screens/ConnectScreen.js [hunk 7]
```diff
@@ -1,17 +1,249 @@
...
+  closeConnectBox() {
+    this.setState({ activeConnection: null });
+  }
+
+  getInitialRegion() {
+    return {
+      latitude: 37.78825,
+      longitude: -122.4324,
+      latitudeDelta: 0.0922,
+      longitudeDelta: 0.0421,
```
Comment (nikitr) [inline src/screens/ConnectScreen.js:107]: instead of hard coding these, will it be based off of the veteran's location when we get that from the backend?

**Review by nikitr — COMMENTED** [review]
this is looking great ken! left some comments/questions in my review.

are we going to take care of the logic for which portion of the map the veteran will see as well as how to handle the loading of all the pins in a future PR?

File: src/components/ConnectPin.js (hunk 5 above)
Comment (nikitr) [reply src/components/ConnectPin.js:30]: can we also make it clear where the user him/herself is located in relation to all pins?

File: src/components/ConnectBox.js (hunk 4 above)
Comment (lbkchen) [reply src/components/ConnectBox.js:48]: hmm yea I think in that case roles would just be undefined and it would go into the else clause

File: src/components/ConnectPin.js (hunk 5 above)
Comment (lbkchen) [reply src/components/ConnectPin.js:30]: yup and yup I will add a current location indicator

File: src/components/ConnectPin.js (hunk 2 above)
Comment (lbkchen) [reply src/components/ConnectPin.js:41]: yup once that is a thing haha

Comment (lbkchen) [reply src/components/ConnectPin.js:41]: need to put those up on aws and maybe we can use imagemagick rails

File: src/screens/ConnectScreen.js (hunk 6 above)
Comment (lbkchen) [reply src/screens/ConnectScreen.js:208]: yup its an object with all the fields and values

File: src/screens/ConnectScreen.js (hunk 7 above)
Comment (lbkchen) [reply src/screens/ConnectScreen.js:107]: yea the deltas might be hard coded still but the lat/lon we will def use the user location

**Comment by lbkchen** [comment]
@nikitr I think the pin rendering algorithm will def be a separate feature/PR and might actually be backend heavy so we just rendering all them pins for now. I realized that switching over to native maps engine (apple maps for ios) is like 5x faster so it may temporarily be OK

File: src/helpers/requesters/BaseRequester.js [hunk 8]
```diff
@@ -56,13 +56,14 @@ class BaseRequester {
       headers: headers,
       body: JSON.stringify(params),
     }).then((response) => {
+      if (!response.ok) { throw response; }
```
Comment (amillman) [inline src/helpers/requesters/BaseRequester.js:59]: does throwing `response.json()` not work here? or does it cause that nested promise problem we had?

File: src/screens/LoginScreen.js [hunk 9]
```diff
@@ -94,9 +91,9 @@ export default class LoginScreen extends React.Component {
         onSuccess && onSuccess(response);
         this.navigateToApp(response);
       }).catch((error) => {
-        console.log(error);
-        onFailure && onFailure(error);
-        this.setState({ errors: error });
+        console.log(error.error);
```
Comment (amillman) [inline src/screens/LoginScreen.js]: remove log

File: src/screens/SignupScreen.js [hunk 10]
```diff
@@ -143,9 +143,9 @@ export default class SignupScreen extends React.Component {
         onSuccess && onSuccess(response);
         this.navigateToApp(response);
       }).catch((error) => {
-        console.log(error);
-        onFailure && onFailure(error);
-        this.setState({ errors: error })
+        console.log(error.errors);
```
Comment (amillman) [inline src/screens/SignupScreen.js]: remove log

File: src/helpers/requesters/BaseRequester.js (hunk 8 above)
Comment (lbkchen) [reply src/helpers/requesters/BaseRequester.js:59]: Yeah same promise problem as before so I have to evaluate it again in the Catch. This is the solution I dug from some guy online who was having similar issues

**Review by nikitr — APPROVED** [review]
LGTM
